test/ansible_processor_test.rb

```diff
@@ -1,8 +1,33 @@
 require "test_helper"
+require "lib/smart_proxy_reports/friendly_message.rb"
 
 class AnsibleProcessorTest < Test::Unit::TestCase
   def setup
     Proxy::Reports::Plugin.settings.stubs(:reported_proxy_hostname).returns("localhost")
+    information = { "result" => {
+      "invocation" => {
+        "module_args" => {},
+      },
+    } }
+    @friendly_instance = FriendlyMessage.new(information)
+  end
+
+  def test_readable_tree_empty
+    assert @friendly_instance.readable_tree([]) == ""
+  end
+
+  def test_readable_tree_nil
+    assert @friendly_instance.readable_tree(nil).nil?
+  end
+
+  def test_normal_tree
+    assert @friendly_instance.readable_tree(["vim", "git", "package", "package2"]) == "vim, git, package, package2"
+  end
+
+  def test_long_tree
+    packages = ["pack1", "pack2", "pack3", "pack4", "pack5", "pack6", "pack7", "pack8", "pack9", "pack10", "pack11", "pack12"]
+    expected = "pack1, pack2, pack3, pack4, pack5, pack6, pack7, pack8, pack9, pack10 and more"
+    assert @friendly_instance.readable_tree(packages) == expected
   end
```

Comment:
Nice set of unit tests, however, these should be present in a separate file: `test/friendly_message_test.rb`. Please move it there.